butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,144 @@
+package com.paypal.butterfly.slack;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.paypal.butterfly.api.TransformationListener;
+import com.paypal.butterfly.api.TransformationRequest;
+import com.paypal.butterfly.api.TransformationResult;
+import org.apache.commons.collections4.map.LRUMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import javax.annotation.PostConstruct;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.io.File;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Observer that notifies the transformation requester
+ * via Slack when a transformation finishes.
+ *
+ * @author edhwang
+ */
+@Configuration
+public class SlackNotifierConfig {
+    private static final Logger logger = LoggerFactory.getLogger(SlackNotifierConfig.class);
+    @Bean
+    public TransformationListener slackNotification() {
+        return new TransformationListener() {
+
+            private String token;
+            private WebTarget target;
+            private String postSuccessJson;
+            private String postAbortJson;
+            private boolean active = false;
+            private Map<String, String> cache;
+
+            @PostConstruct
+            public void postConstruct() {
+                try {
+                    token = System.getProperty("slack.token");
+                    if (token == null) {
+                        logger.error("Unable to locate slack token to send notifications");
+                        return;
+                    }
+                    Path postSuccessPath = new File(this.getClass().getResource("/postSuccess.json").toURI()).toPath();
+                    Path postAbortPath = new File(this.getClass().getResource("/postAbort.json").toURI()).toPath();
+                    postSuccessJson = new String(Files.readAllBytes(postSuccessPath));
+                    postAbortJson = new String(Files.readAllBytes(postAbortPath));
+                    Client client = ClientBuilder.newClient();
+                    target = client.target("https://www.slack.com/api/");
+                    cache = Collections.synchronizedMap(new LRUMap(100));
```

Comment:
One entry in this cache is made of two Strings (key and value). We could consider both combined won't have more than 40 characters in length, each being 2 bytes. So we can say each entry is 80 bytes. A full cache with 100 entries is roughly 8KB then in memory, which is not much. PayPal has around 2000 developers. Each uses Butterfly around 4 times an year. If we don't make this cache big enough for that, the cache will have misses most of the times. Having said that, let's change its size to `1000`. We can afford 80KB in memory for that.